spec/services/wallets/create_interval_wallet_transactions_service_spec.rb

```diff
@@ -0,0 +1,268 @@
+
+      let(:current_date) do
+        DateTime.parse('20 Jun 2022').prev_occurring(created_at.strftime('%A').downcase.to_sym)
+      end
+
+      it 'enqueues a job on correct day' do
+        travel_to(current_date) do
+          create_interval_transactions_service.call
+
+          expect(WalletTransactions::CreateJob).to have_been_enqueued
+            .with(
+              organization_id: customer.organization_id,
+              wallet_id: wallet.id,
+              paid_credits: recurring_transaction_rule.paid_credits,
+              granted_credits: recurring_transaction_rule.granted_credits,
+              source: :interval,
+            )
+        end
+      end
+
+      it 'does not enqueue a job on other day' do
+        travel_to(current_date + 1.day) do
+          expect { create_interval_transactions_service.call }.not_to have_enqueued_job
+        end
+      end
+    end
+
+    context 'when recurring transactions should be created monthly' do
+      let(:interval) { :monthly }
+      let(:current_date) { created_at.next_month }
+
+      it 'enqueues a job on correct day' do
+        travel_to(current_date) do
+          create_interval_transactions_service.call
+
+          expect(WalletTransactions::CreateJob).to have_been_enqueued
+            .with(
+              organization_id: customer.organization_id,
+              wallet_id: wallet.id,
+              paid_credits: recurring_transaction_rule.paid_credits,
+              granted_credits: recurring_transaction_rule.granted_credits,
+              source: :interval,
+            )
+        end
+      end
+
+      it 'does not enqueue a job on other day' do
+        travel_to(current_date + 1.day) do
+          expect { create_interval_transactions_service.call }.not_to have_enqueued_job
+        end
+      end
+
+      context 'when wallet is created on a 31st' do
+        let(:created_at) { DateTime.parse('31 Mar 2021') }
+        let(:current_date) { DateTime.parse('30 Apr 2021') }
+
+        it 'enqueues a job if the month count less than 31 days' do
+          travel_to(current_date) do
+            create_interval_transactions_service.call
+
+            expect(WalletTransactions::CreateJob).to have_been_enqueued
+              .with(
+                organization_id: customer.organization_id,
+                wallet_id: wallet.id,
+                paid_credits: recurring_transaction_rule.paid_credits,
+                granted_credits: recurring_transaction_rule.granted_credits,
+                source: :interval,
+              )
+          end
+        end
+      end
+    end
+
+    context 'when recurring transactions should be created quarterly' do
+      let(:interval) { :quarterly }
+      let(:current_date) { created_at + 3.months }
+
+      it 'enqueues a job on correct day' do
+        travel_to(current_date) do
+          create_interval_transactions_service.call
+
+          expect(WalletTransactions::CreateJob).to have_been_enqueued
+          .with(
```

Comment:
Layout/MultilineMethodCallIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.